### PR TITLE
docs : add SPI device driver usage

### PR DESCRIPTION
**Pull request by manoharahk**
This patch updates SPI driver section in How to Use Peripheral
document.
It describes the low-level SPI device driver implementation
details with examples.

Signed-off-by: Manohara HK <manohara.hk@samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

File: docs/HowToUsePeripheral.md [hunk 1]
```diff
@@ -127,6 +127,119 @@ static struct up_dev_s g_uart0priv = {
...
+
+
+### SPI device driver initialization
+Each SPI device driver must implement the definitions of below prototypes for  
+SPI device initialization
+```
+FAR struct spi_dev_s *up_spiinitialize(int port);
+```
+
+Example implementation is present in *os/arch/arm/src/s5j/s5j_spi.c*
```
Comment (sunghan-chang) [inline docs/HowToUsePeripheral.md]: How about give a link to file ?

Comment (manoharahk) [reply docs/HowToUsePeripheral.md]: It is good to link. I will modify.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**